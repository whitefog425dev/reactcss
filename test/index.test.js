import React from 'react'
import renderer from 'react-test-renderer'
import reactCSS from '../src'

describe('reactCSS', () => {
  test('should return simple css', () => {
    const Component = () => {
      const styles = reactCSS({
        'default': {
          body: {
            backgroundColor: '#fafafa',
          },
        },
      })
      return <div className="body" style={ styles.body } />
    }
    const tree = renderer.create(<Component />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should return multiple css', () => {
    const Component = ({ color }) => {
      const styles = reactCSS({
        'default': {
          title: {
            color,
          },
          card: {
            boxShadow: '0 0 2px rgba(0,0,0,.1)',
          },
        },
      })
      return (
        <div>
          <div className="title" style={ styles.title } />
          <div className="card" style={ styles.card } />
        </div>
      )
    }
    const tree = renderer.create(<Component color="red" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should return complex css', () => {
    const Component = ({ zIndex }) => {
      const styles = reactCSS({
        'default': {
          card: {
            boxShadow: '0 0 2px rgba(0,0,0,.1)',
          },
        },
        'zIndex-2': {
          card: {
            boxShadow: '0 4px 8px rgba(0,0,0,.15)',
          },
        },
      }, { zIndex })
      return (
        <div>
          <div className="card" style={ styles.card } />
        </div>
      )
    }
    const tree = renderer.create(<Component zIndex="2" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
