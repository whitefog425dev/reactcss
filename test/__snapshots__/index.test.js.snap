// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactCSS should return complex css 1`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "MozBoxShadow": "0 4px 8px rgba(0,0,0,.15)",
        "OBoxShadow": "0 4px 8px rgba(0,0,0,.15)",
        "WebkitBoxShadow": "0 4px 8px rgba(0,0,0,.15)",
        "boxShadow": "0 4px 8px rgba(0,0,0,.15)",
        "msBoxShadow": "0 4px 8px rgba(0,0,0,.15)",
      }
    }
  />
</div>
`;

exports[`reactCSS should return multiple css 1`] = `
<div>
  <div
    className="title"
    style={
      Object {
        "color": "red",
      }
    }
  />
  <div
    className="card"
    style={
      Object {
        "MozBoxShadow": "0 0 2px rgba(0,0,0,.1)",
        "OBoxShadow": "0 0 2px rgba(0,0,0,.1)",
        "WebkitBoxShadow": "0 0 2px rgba(0,0,0,.1)",
        "boxShadow": "0 0 2px rgba(0,0,0,.1)",
        "msBoxShadow": "0 0 2px rgba(0,0,0,.1)",
      }
    }
  />
</div>
`;

exports[`reactCSS should return simple css 1`] = `
<div
  className="body"
  style={
    Object {
      "backgroundColor": "#fafafa",
    }
  }
/>
`;
